mod.d/enadiag.yaml

```diff
@@ -0,0 +1,270 @@
+                         "for more details on stat counters."
+          if int(interface.tx_timeout):
+              failure_parts.append("     %s tx_timeout events found.\n" % interface.tx_timeout)
+              failed = 1
+          if int(interface.suspend):
+              failure_parts.append("     %s suspend events found.\n" % interface.suspend)
+              failed = 1
+          if int(interface.wd_expired):
+              failure_parts.append("    %s wd_expired events found.\n" % interface.wd_expired)
+              failed = 1
+          if int(interface.interface_down):
+              warning_parts.append("    %s interface_down events found.\n"
+                                   "    If you have not up/downed the interface this could be a link flap."
+                                  "\n" % interface.interface_down)
+              failed = 0
+          if int(interface.admin_q_pause):
+              failure_parts.append("    %s admin_q_pauses found.\n" % interface.admin_q_pause)
+              failed = 1
+          if int(interface.bw_in_allowance_exceeded):
+              failure_parts.append("    %s bw_in_allowance_exceeded events found.\n"
+                                   % interface.bw_in_allowance_exceeded)
+              failed = 1
+          if int(interface.bw_out_allowance_exceeded):
+              failure_parts.append("    %s bw_out_allowance_exceeded events found.\n"
+                                   % interface.bw_out_allowance_exceeded)
+              failed = 1
+          if int(interface.pps_allowance_exceeded):
+              failure_parts.append("    %s pps_allowance_exceeded events found.\n" % interface.pps_allowance_exceeded)
+              failed = 1
+          if int(interface.conntrack_allowance_exceeded):
+              failure_parts.append("    %s conntrack_allowance_exceeded events found.\n"
+                                   % interface.conntrack_allowance_exceed)
+              failed = 1
+          if int(interface.linklocal_allowance_exceeded):
+              failure_parts.append("    %s linklocal_allowance_exceeded events found."
+                                   % interface.linklocal_allowance_exceeded)
+              failed = 1
+          if failed and warned:
+              failure_string = ""
+              warned_string = ""
+              print(failure_string.join(failure_parts))
+              print(warned.join(warning_parts))
+              print(stat_details)
+          elif failed:
+              failure_string = ""
+              print(failure_string.join(failure_parts))
+              print(stat_details)
+          elif warned:
+              warned_string = ""
+              print(warned.join(warning_parts))
+              print(stat_details)
+          # If we had all 0s for the stats we're checking, success.
+          else:
+              print("[SUCCESS] No ENA problems found on %s." % interface.name)
+      except Exception as ex:
```

Comment:
Similar as earlier: should this fail if a single int constructor fails because for the output format changed?

Reply:
Same as above - if one has changed, they likely all have.